utilities/ride-model-util/src/main/java/com/adobe/ride/utilities/model/ModelObject.java

```diff
@@ -1339,7 +1339,7 @@ public Object generateNodeValue(String parentPath, String key, JSONObject proper
           returnValue = alphaNum;
           break;
         case URI:
-          String uri = DataGenerator.generateURI();
+          String uri = DataGenerator.generateRandomURIRef();
```

Comment:
I'm not super familiar with this aspect of schema, but case URI is no longer distinct from case URI_REF.  Should it be?

Reply:
So technically, the generateRandomURIRef isn't really generating a uri-ref, its generating a uri.  If it made more sense i could instead change the generateURI() method to do what generateRandomURIRef() is currently doing.  

The generateRandomURIRef() method is generating valid uri-ref's, but its only loosely based.  If you see here there are a lot more options for a uri-ref.  That being said, its not really the point of this issue.
https://tools.ietf.org/html/rfc3986#section-4.1

Reply:
Yes, this explains it pretty well -  From the json-schema org: https://json-schema.org/draft-06/json-schema-release-notes.html#formats-uri-vs-uri-reference

As Shay points out, the method `generateRandomURIRef` is not very well named, but I think it's better to correct the URI generation in ModelObject to properly return a spec conforming URI than leave it as is.  I'm inclined to approve.

Reply:
@enortonAtAdobe  The solution for differentiating between the two types for datagen puproses might require a more nuanced and involved solution (i.e. renaming of public methods, which is even more of a backward compatibility break, imho).  

Since a `uri` always qualifies as a `uri-ref` (the inverse is not true), with this change, both of these enums will be getting valid values .  If no change is made, then the `uri` case continues to always get an invalid value.  

@shay-thompson perhaps we can resolve by filing another issue in which we need to generate data for uri-ref's which more closely mirror the idea behind that spec.

Reply:
I would agree with that, however as i'm not using uri-ref i'll leave writing that to you.  I also don't think that should hold up this PR for the issues it resolves.